packages/@lwc/engine-core/src/framework/base-bridge-element.ts

```diff
@@ -104,15 +104,36 @@ function createAttributeChangedCallback(
     };
 }
 
+function createStandardAccessorDescriptor(propName: string) {
+    let prop: any;
+    return {
+        get() {
+            // log warning
+            logWarn(
+                `The property "${propName}" is not publicly accessible. Add the @api annotation to the property declaration or getter/setter in the component to make it accessible.`
+            );
+            return prop;
+        },
+        set(value: any) {
+            // log warning
```

Comment:
Can we log a warning here? I think it's great to warn on both the getter and the setter. 🙂